### ISSUE TITLE
mortality and mortality day 

### ISSUE THREAD

**Issue by liuenqian** [MIT-LCP/mimic-code#789]
### Prerequisites
https://github.com/MIT-LCP/mimic-code/blob/master/concepts/cookbook/mortality.sql

### Description

admittime is accurate to the minute but dod is not. 
subject_id = '8547' and  hadm_id = '131849' and  icustay_id = '200119' 
admissions table admittime: **2173-06-29 01:50:00**   dischtime: **2173-06-30 07:58:00**  deathtime: **2173-06-30 07:58:00**
 icustays table   intime: **2173-06-29 01:50:00**  outtime: **2173-06-30 07:58:00**
 patients table    dod/dod_hosp/dod_ssn: **2173-06-30 00:00:00**
round(((date_part('epoch'::text, (ie.outtime - ie.intime)) / (((60 * 60) * 24))::double precision))::numeric, 4) AS los_icu,
round(((date_part('epoch'::text, (adm.dischtime - adm.admittime)) / (((60 * 60) * 24))::double precision))::numeric, 4) AS los_hospital,

when we use los_icu>2 exclude deceased patient，but patients table‘  dod/dod_hosp/dod_ssn may not accurate. How to exclude those people. 

* what you have tried
* references to similar issues
* queries demonstrating your question (if applicable)

**Comment by JackieMium** [comment]
I can confirm this issue in MIMIC III data.

Carefully look at the patient with `subject_id = '8547'`,there is only one record of hospitalization with `hadm_id = '131849'` and only one ICU icustay with `icustay_id = '200119'`. Thus when limiting an analysis to the first hospitalization/ICU stay, this record will definitely be included.

`SELECT count(*) from icustays where icustay_id = '200119';`:
```
 count 
-------
     1
(1 row)
```

and `SELECT count(*) from admissions where subject_id = '8547';`:

```
 count 
-------
     1
(1 row)
```

Now let's look into more details about this record in `patients` and `admissions`:

` SELECT * from patients where  subject_id = '8547';`:

```
 row_id | subject_id | gender |         dob         |         dod         |      dod_hosp       |       dod_ssn       | expire_flag 
--------+------------+--------+---------------------+---------------------+---------------------+---------------------+-------------
   8085 |       8547 | F      | 2123-05-16 00:00:00 | 2173-06-30 00:00:00 | 2173-06-30 00:00:00 | 2173-06-30 00:00:00 |           1
```

So the patient is deceased according to `expire_flag`, `dod`, `dod_hosp` and `dod_ssn` are the same, `2173-06-30 00:00:00`. Now `icustays`:

` SELECT * from icustays where icustay_id = '200119';`:

```
 row_id | subject_id | hadm_id | icustay_id | dbsource | first_careunit | last_careunit | first_wardid | last_wardid |       intime        |       outtime       |  los   
--------+------------+---------+------------+----------+----------------+---------------+--------------+-------------+---------------------+---------------------+--------
  10883 |       8547 |  131849 |     200119 | carevue  | CCU            | CCU           |           57 |          57 | 2173-06-29 01:50:00 | 2173-06-30 07:58:00 | 1.2556
```

The patient stayed in ICU for 1.2 days (This record **WILL** be incuded if we limit data to records with hospitalizaion to LOS > 1d) and left at `2173-06-30 07:58:00`, **later than** `dod` time point in `patients` table.

Now `admissions`:

`SELECT  subject_id ,hadm_id , admittime, dischtime, deathtime,hospital_expire_flag ,has_chartevents_data from admissions where hadm_id = '131849';`:

```
 subject_id | hadm_id |      admittime      |      dischtime      |      deathtime      | hospital_expire_flag | has_chartevents_data 
------------+---------+---------------------+---------------------+---------------------+----------------------+----------------------
       8547 |  131849 | 2173-06-29 01:50:00 | 2173-06-30 07:58:00 | 2173-06-30 07:58:00 |                    1 |                    1
```

`dischtime` and `deathtime` are the same, `2173-06-30 07:58:00`, also identical to `outtime` in `icustays`.

So the problem here is, if we calculate survival for this patient as `dod` - `intime`, then survival < los_icu, which is apprently wrong.
But when we calculated survival as `deathtime` - `intime`,  then survival = los_icu, which would be right.

In concepts/cookbook/mortality.sql as @liuenqian mentioned, mortality was calculated using both `dod` and `deathtime`. But there was no comparison between the two. This issue may affect the result too.

**Comment by alistairewj** [comment]
I guess this is a consequence of storing a date in a `DATETIME` column. Probably easiest just to truncate everything to a date and keep LOS as an integer!

**Comment by JackieMium** [comment]
I've managed to capture some obvious errors here (at least I thought).

1. subject_id = 14288, hadm_id = 106324, icustay_id = 295882

` SELECT * from patients where subject_id = 14288;`

```
 row_id | subject_id | gender |         dob         |         dod         | dod_hosp |       dod_ssn       | expire_flag 
--------+------------+--------+---------------------+---------------------+----------+---------------------+-------------
  13510 |      14288 | F      | 2111-06-05 00:00:00 | 2194-05-19 00:00:00 | [NULL]   | 2194-05-19 00:00:00 |           1
```

`dod` is **2194-05-19.**

`SELECT subject_id, hadm_id, admittime, dischtime, deathtime, hospital_expire_flag from admissions where subject_id = 14288;`

```
 subject_id | hadm_id |      admittime      |      dischtime      | deathtime | hospital_expire_flag 
------------+---------+---------------------+---------------------+-----------+----------------------
      14288 |  106324 | 2194-05-22 05:38:00 | 2194-05-29 11:31:00 | [NULL]    |                    0
```

`dischtime` and `deathtime` are **2194-05-29**, 10 days later as indicated by `dod`, `admittime` is **2194-05-22**, 3 days later than `dod`.

`SELECT subject_id, hadm_id, icustay_id, dbsource, intime, outtime, los from icustays where subject_id = 14288;`

```
 subject_id | hadm_id | icustay_id | dbsource |       intime        |       outtime       |  los   
------------+---------+------------+----------+---------------------+---------------------+--------
      14288 |  106324 |     295882 | carevue  | 2194-05-22 19:30:00 | 2194-05-23 14:58:00 | 0.8111
```

`intime` and `outtime` are, however, between `admittime` and `dischtime`. Still later than `dod`.

Could `dod` be typo where `2194-05-29` was accidently inputed as `2194-05-19`?

----

2. subject_id = 10518, ham_id = 140994, icustay_id = 298034

`SELECT subject_id, hadm_id, icustay_id, dbsource, intime, outtime, los from icustays where subject_id = 10518;`

```
 subject_id | hadm_id | icustay_id | dbsource |       intime        |       outtime       |  los   
------------+---------+------------+----------+---------------------+---------------------+--------
      10518 |  140994 |     298034 | carevue  | 2189-10-22 19:31:31 | 2189-10-24 19:21:13 | 1.9928
```

`SELECT * from patients where subject_id = 10518;`

```
 row_id | subject_id | gender |         dob         |         dod         |      dod_hosp       |       dod_ssn       | expire_flag 
--------+------------+--------+---------------------+---------------------+---------------------+---------------------+-------------
   9954 |      10518 | M      | 2131-12-14 00:00:00 | 2189-10-21 00:00:00 | 2189-10-21 00:00:00 | 2189-10-25 00:00:00 |           1
```

`SELECT subject_id, hadm_id, admittime, dischtime, deathtime, hospital_expire_flag from admissions where subject_id = 10518;`

```
 subject_id | hadm_id |      admittime      |      dischtime      |      deathtime      | hospital_expire_flag 
------------+---------+---------------------+---------------------+---------------------+----------------------
      10518 |  140994 | 2189-10-22 18:02:00 | 2189-10-25 10:20:00 | 2189-10-21 10:20:00 |                    1
```

Similar as case 1 here but cannot be simply expained as a typo.

**Comment by alistairewj** [comment]
If you check the diagnosis field in admissions, does it mention anything
about them being an organ donor? There were around 30 cases .. probably
should have been removed from the database.

>

**Comment by JackieMium** [comment]
> If you check the diagnosis field in admissions, does it mention anything about them being an organ donor? 

Yes of course I double checked that. That why I copied long text from `noteevents` about the 2 cases above. The two cases are regular hospitalization: The first case, a 82yr old female got to emergency dept due to sudden a stroke in brain and later discharged to hospice, no mention of organ donor in `noteevents` or `admissions`. the second acse, 57yr old male got to ED due to a siezure, later transferred to hospipice service on patient's will. No mention of of organ donor found either.
